Feedback widget v3

casthewiz opened the pull request:
### Stories/Links:

DOP-2400

### Current Behavior:

_Put a link to current staging or production behavior, if applicable_

### Staging Links:

[Realm](https://docs-mongodbcom-integration.corp.mongodb.com/master/realm/cassidy.schaufele/feedback-widget-v3c/)
[Cloud](https://docs-mongodbcom-integration.corp.mongodb.com/master/cloud-docs/cassidy.schaufele/feedback-widget-v3a/)

### Notes:

Culmination of multiple PRs for V3 of the feedback widget.
My changes are minor (changing our modal screenshot transition logic and making element access null safe, and also minor style changes to reconcile with leafygreen dependency bumps).

Feedback currently directs to a test slack channel.

sandynguyenn reviewed (changes requested):
Hi Cassidy! Things are looking great, but I have a couple of small nits:

Is it possible to increase the width of the feedback widget so that the negative feedback button and the suggestion button don't span two lines? The spacing seems off balance with the current width.
![Screen Shot 2022-11-15 at 2 12 42 PM](https://user-images.githubusercontent.com/116376420/202006236-2a00de31-6885-4216-bbd3-62a056b9c7df.png)
This was my original design where the width spanned 234px, It doesn't have to be 234px but the button text should only span one line:
![Screen Shot 2022-11-15 at 2 20 46 PM](https://user-images.githubusercontent.com/116376420/202006985-a1421409-a45c-4d9a-9642-7f49ecafe3ea.png)

The second thing was that the email input font is too dark, users may misinterpret it as already inputted text rather than placement text. It should be the same font color/have the same styling as the "How did this page help you?" text for consistency:
![Screen Shot 2022-11-15 at 2 13 02 PM](https://user-images.githubusercontent.com/116376420/202007252-0093239f-c87d-4010-b0c0-9b79ab2d0e60.png)

Let me know if you have any questions! Thanks :D

casthewiz commented:
Sure thing - I think the card bump squished the padding. I'll adjust that and the input font.

rayangler commented on src/components/Widgets/FeedbackWidget/components/ScreenshotButton.js:
```diff
@@ -1,35 +1,346 @@
+    for (const elem in listOfElements) {
+      if (
+        elem &&
+        (listOfElements[elem]?.className?.includes('SidenavContainer') ||
+          listOfElements[elem]?.className?.includes('StyledHeaderContainer'))
```
I'm not sure if this code path will ever hit production. These seem pretty specific to styled component class names, which only appear in development. 

I'm okay with keeping as-is, but maybe a comment regarding this might be useful for context. I'm also unsure as to why we need to differentiate between fixed of absolute for these two elements 🤔

seungpark replied:
im guessing this is due to the nature of sidenav or header where they are fixed in position and calculated from body? but i second the `not true in production`. seems the `absolute` positioning works for sidenav and header components in staged link

rayangler reviewed (commented):
Tried to keep my review focused more on the recent changes since most of it should have been reviewed before, but I still wanted to ask about or point out some things that stood out prior to merging.

rayangler commented on src/components/Widgets/FeedbackWidget/FeedbackHeading.js:
```diff
@@ -1,16 +1,18 @@
       <>
         <StarRating size="lg" />
-        <StarRatingLabel>Give Feedback</StarRatingLabel>
+        <StarRatingLabel>Share Feedback</StarRatingLabel>
```
Oh, actually, just saw #705 should handle this. Maybe we should merge that first?

casthewiz replied:
🤦 Totally my mistake, it would be very helpful indeed if that was merged first. Thank you for flagging!

sandynguyenn commented:
Sorry another nit, but the text here doesn't seemed to be middle aligned anymore with the modal:

![Screen Shot 2022-11-17 at 10 06 24 AM](https://user-images.githubusercontent.com/116376420/202482515-bf20c1eb-be22-4845-923d-d6d9ffd4b604.png)

allisonmui9 reviewed (approved):
Small nit: Can we update the select screenshot font in the frame? It's using the old font. Here's an updated file. 
![select an element](https://user-images.githubusercontent.com/57725481/202516236-09e2bec8-2beb-4a24-b4d9-d35f740a8712.svg)

sandynguyenn reviewed (changes requested):
Will approve once Allison's comment + my comment above have been resolved!

casthewiz commented:
@sandynguyenn https://docs-mongodbcom-integration.corp.mongodb.com/master/realm/cassidy.schaufele/feedback-widget-v3/ has the latest round of changes. I've adjusted padding in various elements and removed several styles that seemed to be redundant/conflicting - net result is that everything should be properly centered, if not as close to centered as I can get it currently. 

SVG is also swapped, so the element selection is on the latest branding. Thanks!

casthewiz commented:
I've since re-staged latest adjustments at https://docs-mongodbcom-integration.corp.mongodb.com/master/realm/cassidy.schaufele/feedback-widget-v3c/ .

sandynguyenn reviewed (approved):
lgtm

seungpark reviewed (approved):
functionally LGTM. 

seems like there's some clean up to do with styling. generally removing important flags and using more flexboxes with a set container, using more theme variables.

seungpark commented:
> functionally LGTM.
> 
> seems like there's some clean up to do with styling. generally removing important flags and using more flexboxes with a set container, using more theme variables.

also a minor improvement (non blocking) to prevent re-rendering of all four "sides" of the overlay. one element with borders (possibly could be transparent borders to give off same feel):

![image](https://user-images.githubusercontent.com/13054820/203142876-66f1d12d-36a3-49c6-b336-8215990ef897.png)

casthewiz commented on src/components/Widgets/FeedbackWidget/components/ScreenshotButton.js:
```diff
@@ -1,35 +1,346 @@
+    if (!isFWSelected(listOfElements)) {
+      domElement = listOfElements[0];
+      for (const elem in listOfElements) {
+        if (!listOfElements[elem]?.className?.includes('overlay')) {
```
Resolved post-merge!
